src/completion.ts

```diff
@@ -53,13 +62,15 @@ export function asCompletionItem(
         },
     };
 
+    const line = document.getLine(position.line);
```

Comment:
This is fairly expensive and is being done for every single completion while it could be done once, earlier since the result is not completion-specific.